From b43f24cc8128d96e1a6169622c4e5828995634b2 Mon Sep 17 00:00:00 2001
Message-Id: <b43f24cc8128d96e1a6169622c4e5828995634b2.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Wed, 14 Jan 2015 01:19:44 +0530
Subject: [PATCH 48/95] MUST_REBASE [VPG]: drivers/video/adf: Changes in dsi
 get hw state for dual link

Changes for DSI dual link HW state if BIOS enables the dual link panel.
Changes to check the dsi video mode for the dual link panel.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

Issue: GMINL-5276
Change-Id: I8481f07a58f979f3d27baa0d83a5431f6a18bbee
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/chv_dpio.c     |   11 ++++++---
 drivers/video/adf/intel/core/vlv/vlv_pipeline.c |   28 ++++++++++++++++++-----
 2 files changed, 30 insertions(+), 9 deletions(-)

diff --git a/drivers/video/adf/intel/core/vlv/chv_dpio.c b/drivers/video/adf/intel/core/vlv/chv_dpio.c
index 57ae8a9..e9af212 100644
--- a/drivers/video/adf/intel/core/vlv/chv_dpio.c
+++ b/drivers/video/adf/intel/core/vlv/chv_dpio.c
@@ -55,12 +55,17 @@ static inline struct vlv_dc_config *get_vlv_dc_config(
 
 int vlv_disp_to_port(struct vlv_pipeline *pipeline)
 {
-	struct vlv_pll *pll = &pipeline->pll;
-	enum port port_id;
+	struct dsi_pipe *dsi_pipe = NULL;
+	struct dsi_context *dsi_ctx = NULL;
+	enum port port;
+	enum port port_id = PORT_A;
 
 	switch (pipeline->type) {
 	case INTEL_PIPE_DSI:
-		port_id = pipeline->port.dsi_port[pll->port_id].port_id;
+		dsi_pipe = &pipeline->gen.dsi;
+		dsi_ctx = &dsi_pipe->config.ctx;
+		for_each_dsi_port(port, dsi_ctx->ports)
+			port_id = pipeline->port.dsi_port[port].port_id;
 		break;
 	case INTEL_PIPE_HDMI:
 		port_id = pipeline->port.hdmi_port.port_id;
diff --git a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
index f3b3fdc..0bb251a 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_pipeline.c
@@ -530,12 +530,20 @@ bool vlv_is_vid_mode(struct intel_pipeline *pipeline)
 {
 	struct vlv_pipeline *disp = to_vlv_pipeline(pipeline);
 	struct vlv_dsi_port *dsi_port = NULL;
-	struct vlv_pll *pll = &disp->pll;
+	struct dsi_pipe *dsi_pipe = NULL;
+	struct dsi_context *dsi_ctx = NULL;
+	enum port port;
 	bool ret = false;
 
 	if (disp->type == INTEL_PIPE_DSI) {
-		dsi_port = &disp->port.dsi_port[pll->port_id];
-		ret = vlv_dsi_port_is_vid_mode(dsi_port);
+		dsi_pipe = &disp->gen.dsi;
+		dsi_ctx = &dsi_pipe->config.ctx;
+		for_each_dsi_port(port, dsi_ctx->ports) {
+			dsi_port = &disp->port.dsi_port[port];
+			ret = vlv_dsi_port_is_vid_mode(dsi_port);
+			if (ret)
+				return ret;
+		}
 	}
 
 	return ret;
@@ -545,12 +553,20 @@ bool vlv_can_be_disabled(struct intel_pipeline *pipeline)
 {
 	struct vlv_pipeline *disp = to_vlv_pipeline(pipeline);
 	struct vlv_dsi_port *dsi_port = NULL;
-	struct vlv_pll *pll = &disp->pll;
+	struct dsi_pipe *dsi_pipe = NULL;
+	struct dsi_context *dsi_ctx = NULL;
+	enum port port;
 	bool ret = false;
 
 	if (disp->type == INTEL_PIPE_DSI) {
-		dsi_port = &disp->port.dsi_port[pll->port_id];
-		ret = vlv_dsi_port_can_be_disabled(dsi_port);
+		dsi_pipe = &disp->gen.dsi;
+		dsi_ctx = &dsi_pipe->config.ctx;
+		for_each_dsi_port(port, dsi_ctx->ports) {
+			dsi_port = &disp->port.dsi_port[port];
+			ret = vlv_dsi_port_can_be_disabled(dsi_port);
+			if (ret)
+				return ret;
+		}
 	}
 
 	return ret;
-- 
1.7.9.5

